app/models/user.rb

```diff
@@ -14,7 +14,8 @@ def password_complexity
 
     # Passwords must have characters from at least two groups, identified by these regexes (last one is punctuation)
     matches = [/[a-z]/, /[A-Z]/, /[0-9]/, /[^\w\s]/].select { |rx| rx.match(password) }.size
-    errors.add :password, 'must include characters from at least two groups (lower case, upper case, numbers, special characters)' unless matches >= 2
+    errors.add :password, 'must include characters from at least three groups (lower case, upper case, numbers, special characters)' unless matches >= 3
+    errors.add :password, 'must be ten characters or more in length' unless password.length >= 10
```

Comment:
Normally password length would be configured in config/initializers/devise.rb, look for

    # Range for password length.
    config.password_length = 8..128

Reply:
Agreed, we also are using devise-security, so I think the "must include characters from at least three" needs to be moved out of here as well.

Reply:
I actually looked into this and I don't think we can get rid of our RegEx and meet the intent of "characters from at least two groups" with Devise out-of-the-box configuration in the `config do` block.

Reply:
https://github.com/SaraAlert/SaraAlert/blob/master/config/initializers/devise-security.rb#L11

Reply:
Right, but I don't think you can say "characters from at least two groups" with that construct

Reply:
Right, but looking at the implementation it looks like you can add additional categories via options - looking in to this now.